src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { type FC } from 'react';
+
+import searchIcon from './assets/search-icon.svg';
+import { DEFAULT_PLACEHOLDER } from './SearchBar.constants';
+import { useSearchBar } from './SearchBar.hook';
+import { SearchContainer, StyledInput, SearchIconButton } from './SearchBar.styles';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const SearchBar: FC<SearchBarProps> = (props) => {
```

Comment:
The SearchBar component isn't wrapped with React.memo. For a small presentational component that receives handlers from parents, consider wrapping it with React.memo to avoid unnecessary re-renders when parent state changes unrelated to this component.

**Sugerencia:**
```typescript
Create the component as a const and export a memoized version, e.g.:
const SearchBarComponent: FC<PropsSearchBar> = (props) => { ... };
export const SearchBar = React.memo(SearchBarComponent);
```